src/ui/citation-popup/view.js

```diff
@@ -1,22 +1,107 @@
+
+/* State */
+const refs = JSON.parse(inputRefsView.textContent);
+const refsMap = new Map();
+refs.forEach(ref => refsMap.set(ref["id"], ref));
+const state = {
+    selectedRefs: new Set()
+};
+
+configAutoComplete();
 
 /* Event Listeners */
+selectedRefsView.addEventListener("click", event => {
+    if (event.target.classList.contains("icon_remove")) {
+        removeReference(event.target.parentNode.id);
+    }
+});
 
-// When selecting a certain reference, set the "reference_id" field accordingly
-let selectedReferenceView = null;
-for (let i = 0; i < refs.length; i++) {
-    let ref = refs[i];
-    ref.addEventListener("click", event => {
-        referenceIdInput.value = event.target.id;
-        
-        // Attach some css style
-        if (selectedReferenceView !== null) {
-            selectedReferenceView.className = "";
+function configAutoComplete () {
+
+    const autoCompleteJS = new autoComplete({
+        placeHolder: "Search for references...",
+        data: {
+            src: refs,
+            keys: ["title"],
+            filter: list => {
+                return list.filter(item => !state.selectedRefs.has(item.value["id"]));
+            }
+        },
+        resultsList: {
+            noResults: true,
+            maxResults: 15,
+            tabSelect: true
+        },
+        resultItem: {
+            element: (item, data) => {
+                // Modify Results Item Style
+                item.style = "display: flex; justify-content: space-between;";
+                // Modify Results Item Content
+                item.innerHTML = `
+                    <span style="text-overflow: ellipsis; white-space: nowrap; overflow: hidden;">
+                        ${data.match}
+                    </span>
+                `;
+            },
+            highlight: true
+        },
+        events: {
+            input: {
+                focus: () => {
+                    if (autoCompleteJS.input.value.length) autoCompleteJS.start();
+                }
+            }
         }
+    });
+    autoCompleteJS.searchEngine = "loose";
 
-        selectedReferenceView = event.target;
-        selectedReferenceView.className = "selected";
+    autoCompleteJS.input.addEventListener("selection", event => {
+        const feedback = event.detail;
+        const selection = feedback.selection.value;
+        addReference(selection["id"]);
     });
+
+}
+
+function addReference (refId = "") {
+    state.selectedRefs.add(refId);
+    render();
+}
+
+function removeReference (refId = "") {
+    state.selectedRefs.delete(refId);
+    render();
+}
+
+/* Rendering state-based UI */
+function render () {
+    const selectedRefsArray = Array.from(state.selectedRefs);
+    selectedRefsView.innerHTML = template(selectedRefsArray);
+    output.value = JSON.stringify(selectedRefsArray);
+}
+
+/**
+ * Returns an HTML representation of an array of refs
+ * @param {Reference[]} refs
+ * @returns string
+ */
+function template (refs = []) {
```

Comment:
That's a good idea to do all the rendering in one place like this. Ideally a lib like React would be better but I can agree it's a bit overkill at this stage.